lib/zclient.c

```diff
@@ -1991,6 +1991,35 @@ int lm_label_manager_connect(struct zclient *zclient)
 	return (int)result;
 }
 
+/*
```

Comment:
Had to be split to avoid blocking bgpd on label request. Not entirely cut/paste because of zclient_send_message() vs. writen(). Time pressure prevented me from fully analyzing implications of substituting zclient_send_message() into lm_get_label_chunk(), and I also don't have a good fixture for testing ldpd. Didn't want to break something in ldpd by accident.

Reply:
Yes, changing that would break ldpd badly. It's not prepared to allocate labels asynchronously, a few changes in the label distribution engine would be necessary (this is on my to-do list now).